rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+from rlberry.envs import Chain
+from rlberry.envs.benchmarks.ball_exploration import PBall2D
+from rlberry.manager import AgentManager
+import numpy as np
+from rlberry.seeding import set_external_seed
+import tempfile
+import os
+
+SEED = 42
+
+
+def check_agent_manager(Agent, continuous_state=False):
+    """
+    Check that the agent is compatible with :class:`~rlberry.manager.AgentManager`.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    try:
+        agent = AgentManager(
+            Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+        )
+        agent.fit()
```

Comment:
Suggestion: also test eval

```python
agent.fit()
agent.eval_agents()
```

Reply:
same reservation as above.

Reply:
It's just to check that `eval` is also running. If I remember correctly, a call to `fit` will only store the agent instances in disk (`dump` method of `AgentHandler`), whereas we need to load the agent from the disk (`AgentHandler.load`) to make `eval_agents` work. Sometimes `dump` works, but not `load`.

Reply:
`load` is now checked in `check_save` (line 184 of `check_agent.py`). Maybe I can change the name to `check_save_load`

Reply:
but yeah, I can add it here, there is no such thing as too much testing :)

Reply:
Ah, there is a problem with eval_agent that takes too much time. For now I will not use it.